fiona/ogrext.pyx

```diff
@@ -895,6 +883,15 @@ cdef class WritingSession(Session):
             name_b = collection.name.encode('utf-8')
             name_c = name_b
 
+            for k, v in kwargs.items():
+                k = k.upper().encode('utf-8')
+                if isinstance(v, bool):
+                    v = ('ON' if v else 'OFF').encode('utf-8')
+                else:
+                    v = str(v).encode('utf-8')
+                log.debug("Set option %r: %r", k, v)
+                options = CSLAddNameValue(options, <const char *>k, <const char *>v)
+
```

Comment:
We don't have a shim for `GDALDatasetCreateLayer`, so the option encoding goes here.